### ISSUE TITLE
remote station gets all messed up on intermittent connectivity

### ISSUE THREAD

**Issue by Angel996** [tmux/tmux#1296]
I use tmux to monitor my remote mining rigs. Recently I've had some internet connectivity problems at my local station which I use to monitor my rigs (router/isp problems, that can happen, right?). The thing is, connection was not terminated, it was a kind of poor connection that lingered on, but could not receive text sent by tmux fast enough to catch up with it.

Turns out, when tmux cannot push all text generated by remote station to my local monitoring computer, it messes up all apps running on the remote station, making it go into some weird state where programs terminate randomly, videocards overheat, all goes to hell, which is a nightmare to me.

Is there a way to specify some kind of timeout, so that when tmux cannot transmit program output to remote session, it would just detach automatically and allow the programs that run on remote station to do whatever they have to? Or, should I put it that way, is there a way of telling tmux that running programs is actually a top priority, not trying to push text thru an internet connection which barely works?

I've searched thru man pages but haven't found an option for this.

thanks!

**Comment by krader1961** [comment]
Define "messes up all apps running on the remote station". And how could tmux cause the video card to overheat? I used to get paid to solve problems like yours and am quite confident you've misdiagnosed the problem. Consider that if you run a program which generates output continuously inside a tmux session and no client is connected to that session that program will not experience any problems. The tmux server will simply buffer a finite amount of data as it updates the virtual screen. That is analogous to the situation you describe where the connection between the tmux client and server has a very low bandwidth.

**Comment by Angel996** [comment]
I didn't feel I need to go into details, but here goes, if it's necessary. GPU overheating is due to the fact that miner app gets into a weird state where it cannot function normally, it just locks up causing GPUs to heat up. However, fan control shell script, which is supposed to accelerate fans on temperature rise, fails to to so due to tmux blocking all output: it simply can't printf its messages. Thus, temps go up in uncontrolled manner.

All this happens whenever my local SSH client cannot process data, either due to some network congestion or even some misbehaved program locking up my local computer (which happened today). SSH windows froze and remote stations just went wild (which I could see when connection was restored).

Anyway, it's all up to tmux and my question was: how to set it to detach when it's text buffer is overflown instead of messing up computers it is run on?

**Comment by krader1961** [comment]
No, it is not "all up to tmux". Try this simple experiment. Start a new tmux session: `tmux new-session -A -s 'testing'`. In a different terminal run `top` or `htop` (or your preferred process monitoring app). Generate a huge stream of unique lines in that session: `seq 99999999999`. Notice that the tmux server process and the `seq` process are close to being CPU bound. Suspend the tmux session by pressing your tmux command prefix (usually [ctrl-A] or [ctrl-B]) then [ctrl-Z]. Notice that the tmux server and `seq` processes are still running. After a few seconds type `fg` to bring the tmux client back into the foreground. Notice that there is a big jump in the sequence numbers reflecting the fact that `seq` was still running while the tmux server was unable to deliver that text to the tmux client.

In other words, tmux by design already does what you want. Having said that you didn't mention which version of the tmux server is running on your remote systems. I suppose it's possible you're running a really old version that doesn't handle this scenario correctly.

**Comment by Angel996** [comment]
The scenario you described does not actually reflect my situation. Like I said, tmux runs fine if connection is terminated: network cable unplugged, client window closed, CTRL-B/d pressed, connection terminated, etc etc. Remote stations goes on and all works fine, and I can reconnect later by SSHing onto my remote station and typing _tmux attach_. Problems arise when connectivity/bandwidth is limited: there is still an active TCP connection, but packets do not travel as fast as they should or they are delayed significantly. In this case tmux *thinks* it is connected and tries to send data, but data cannot actually go thru. In this case all programs sending their output to tmux lock up or even get terminated.

I'm not sure why this happens, I'm just describing whatever goes on.

**Comment by nicm** [comment]
This issue does not include enough information. Please read the [CONTRIBUTING](https://raw.githubusercontent.com/tmux/tmux/master/CONTRIBUTING) file.
